packages/docusaurus-theme-search-algolia/src/index.ts

```diff
@@ -57,21 +63,32 @@ export default function themeSearchAlgolia(context: LoadContext): Plugin<void> {
     },
 
     async contentLoaded({actions: {addRoute}}) {
+      if (isSearchPageDisabled) {
+        return;
+      }
+
       addRoute({
-        path: normalizeUrl([baseUrl, 'search']),
+        path: normalizeUrl([baseUrl, searchPagePath as string]),
         component: '@theme/SearchPage',
         exact: true,
       });
     },
 
     async postBuild({outDir}) {
+      if (isSearchPageDisabled) {
+        return;
+      }
+
+      const siteUrl = normalizeUrl([url, baseUrl]);
+
       try {
         fs.writeFileSync(
           path.join(outDir, OPEN_SEARCH_FILENAME),
           renderOpenSearchTemplate({
             title,
-            url: url + baseUrl,
-            favicon: favicon ? normalizeUrl([url, baseUrl, favicon]) : null,
+            siteUrl,
+            searchUrl: normalizeUrl([siteUrl, searchPagePath as string]),
```

Comment:
not a fan of those "as string".

If you use `if (searchPagePage) { }` instead then TS can know inside it can't be a false | null so this becomes unneeded